pkg/skaffold/build/local/types.go

```diff
@@ -91,10 +92,15 @@ func NewBuilder(cfg Config) (*Builder, error) {
 		return nil, fmt.Errorf("getting localCluster: %w", err)
 	}
 
+	imageLoadingRequired, err := isImageLoadingRequired(cfg.GlobalConfig())
+	if err != nil {
+		return nil, fmt.Errorf("getting imageLoadingRequired: %w", err)
+	}
+
 	var pushImages bool
 	if cfg.Pipeline().Build.LocalBuild.Push == nil {
-		pushImages = !localCluster
-		logrus.Debugf("push value not present, defaulting to %t because localCluster is %t", pushImages, localCluster)
+		pushImages = !localCluster && !imageLoadingRequired
```

Comment:
does't `!localCluster && !imageLoadingRequired` always evaluate to `false` for `localCluster=true`?
I don't think it's possible to capture all cases between `minikube`, `kind` and `remote` clusters using these two variables. 

How about a:
```go
type Cluster struct {
	IsLocal    bool
	PushImages bool
	LoadImages bool
}
```
and a `config.GetCluster() (Cluster, error)` function instead of the `GetLocalCluster` and `ImageLoadingReqd` functions

Reply:
Yes, it seems like a leaky abstraction. I'll work on adding `config.GetCluster`.

Reply:
@gsquared94 introduced config.GetCluster.